Sources/GRPCCore/GRPCClient.swift

```diff
@@ -0,0 +1,823 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A gRPC client.
+///
+/// A ``GRPCClient`` communicates to a server via a given ``ClientTransport``.
+/// You can start RPCs to the server by calling the corresponding method:
+/// - ``unary(request:descriptor:serializer:deserializer:handler:)``
+/// - ``clientStreaming(request:descriptor:serializer:deserializer:handler:)``
+/// - ``serverStreaming(request:descriptor:serializer:deserializer:handler:)``
+/// - ``bidirectionalStreaming(request:descriptor:serializer:deserializer:handler:)``
+///
+/// You can set ``MethodConfiguration``s on this client to override whatever configurations have been
+/// set on the given transport.
+/// You can also use ``ClientInterceptor``s to implement cross-cutting logic which apply to all
+/// RPCs. Example uses of interceptors include authentication and logging.
+///
+/// ## Creating and configuring a client
+///
+/// The following example demonstrates how to create and configure a server.
+///
+/// ```swift
+/// // Create and add an in-process transport.
+/// let inProcessTransport = InProcessClientTransport()
+/// let client = GRPCClient(transport: inProcessTransport)
+///
+/// // Create and add some interceptors.
+/// client.interceptors.add(StatsRecordingServerInterceptors())
+///
+/// // Create and add some method configurations.
+/// let defaultConfiguration = MethodConfiguration(
+///     executionPolicy: ...,
+///     timeout: ...
+/// )
+/// let registry = MethodConfigurationRegistry(defaultConfiguration: defaultConfiguration)
+/// client.methodConfigurationOverrides = registry
+/// ```
+///
+/// ## Starting and stopping the client
+///
+/// Once you have configured the client, call ``run()`` to start it. Calling ``run()`` connects to the given
+/// transport.
+///
+/// ```swift
+/// // Start running the client.
+/// try await client.run()
```

Comment:
Since this method won't return IMO it might be worth slightly extending the example to call run in a child task and use the client to execute an example request